sdk/resource/os.go

```diff
@@ -19,7 +19,7 @@ import (
 	"strings"
 
 	"go.opentelemetry.io/otel/attribute"
-	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
```

Comment:
The `osTypeAttributeMap` map should be updated. E.g. `aix` and `zos` values could be handled as well.

References:

- https://github.com/golang/go/blob/master/src/go/build/syslist.go
- https://github.com/open-telemetry/semantic-conventions/blob/v1.21.0/docs/resource/os.md